src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -460,516 +886,228 @@ private static int GetFrequency(double value)
-                    Float value = retTopics[i].Value;
-                    Contracts.Assert(value >= 0);
-                    Contracts.Assert(0 <= index && index < len);
-                    if (count < len)
+                    for (int i = 0; i < _exes.Length; i++)
                     {
-                        Contracts.Assert(i == 0 || indices[i - 1] < index);
-                        indices[i] = index;
+                        getters[i](ref src);
+                        docSizeCheck[i] += states[i].FeedTrain(Host, in src);
                     }
-                    else
-                        Contracts.Assert(index == i);
-
-                    values[i] = value;
-                    normalizer += value;
                 }
-
-                if (normalizer > 0)
+                for (int i = 0; i < _exes.Length; i++)
                 {
-                    for (int i = 0; i < count; i++)
-                        values[i] = (Float)(values[i] / normalizer);
+                    Host.Assert(corpusSize[i] == docSizeCheck[i]);
+                    states[i].CompleteTrain();
                 }
-                dst = new VBuffer<Float>(len, count, values, indices);
             }
+        }
 
-            public void Dispose()
-            {
-                _ldaTrainer.Dispose();
-            }
+        protected override IRowMapper MakeRowMapper(ISchema schema)
+        {
+            return new Mapper(this, Schema.Create(schema));
         }
+    }
 
-        private ColumnType[] InitColumnTypes(int numTopics)
+    /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+    public sealed class LdaEstimator : IEstimator<LdaTransform>
+    {
+        internal static class Defaults
         {
-            Host.Assert(Utils.Size(Infos) > 0);
-            var types = new ColumnType[Infos.Length];
-            for (int c = 0; c < Infos.Length; c++)
-                types[c] = new VectorType(NumberType.Float, numTopics);
-            return types;
+            public const int NumTopic = 100;
+            public const Single AlphaSum = 100;
+            public const Single Beta = 0.01f;
+            public const int Mhstep = 4;
+            public const int NumIterations = 200;
+            public const int LikelihoodInterval = 5;
+            public const int NumThreads = 0;
+            public const int NumMaxDocToken = 512;
+            public const int NumSummaryTermPerTopic = 10;
+            public const int NumBurninIterations = 10;
+            public const bool ResetRandomGenerator = false;
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        private readonly IHost _host;
+        private readonly LdaTransform.ColumnInfo[] _columns;
+
+        /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+        /// <param name="env">The environment.</param>
+        /// <param name="inputColumn">The column containing text to tokenize.</param>
```

Comment:
>The column containing text to tokenize. [](start = 38, length = 39)

how does that compute with the fact that you expect a fixed vector of floats as input?

Reply:
I'm not sure I understand. 'Tokens' are strings, right? We expect a float vector, so it must be the word bag or something like that, right?

---
In reply to: [232501122](https://github.com/dotnet/machinelearning/pull/1410#discussion_r232501122) [](ancestors = 232501122,232438000)

Reply:
fixed

---
In reply to: [233588116](https://github.com/dotnet/machinelearning/pull/1410#discussion_r233588116) [](ancestors = 233588116,232501122,232438000)